src/core/utils/parse_cmdline.cpp

```diff
@@ -50,21 +52,17 @@ otError CmdLineParser::ParseCmd(char *aString, uint8_t &aArgc, char *aArgv[], ui
 
     aArgc = 0;
 
-    for (cmd = aString; IsSpaceOrNewLine(*cmd) && *cmd; cmd++)
-        ;
-
-    if (*cmd)
+    for (cmd = aString; *cmd; cmd++)
     {
-        aArgv[aArgc++] = cmd++; // the first argument
-    }
-
-    for (; *cmd; cmd++)
-    {
-        if (IsSpaceOrNewLine(*cmd))
+        if ((*cmd == '\\') && (*(cmd + 1) == ' '))
```

Comment:
Should we also escape `\` itself in case a string argument ends with `\`, but there is another argument follows?

Reply:
Agree, will do.

@bukepo  @simonlingoogle 
Should we escape other separator characters as well (`'\t'`, `'\r'`, `'\n'`)?

Reply:
I think it would be good to support other white spaces too. And maybe also `\\`?

Reply:
@simonlingoogle take a look to the latest commits. We can extend `IsEscapable()` as desired.

Reply:
Do you mean `\t` or `\<tab>`? 
I think we can implement `IsEscapable` by calling `IsSpaceOrNewLine` directly:
```
return IsSpaceOrNewLine(c) || c == '\\'
```

Reply:
@simonlingoogle yes, I was thinking about that. Will do.